R/derive_var_extreme_date.R

```diff
@@ -0,0 +1,547 @@
+#' Derive Last Known Alive Date
+#'
+#' @description
+#' `r lifecycle::badge("deprecated")`
+#'
+#' *Deprecated*, please use `derive_var_extreme_dt()` instead. Add the last
+#' known alive date (`LSTALVDT`) to the dataset.
+#'
+#' @param dataset Input dataset
+#'
+#'   The variables specified by `subject_keys` are required.
+#'
+#' @param source_datasets A named `list` containing datasets in which to search for the
+#'   last known alive date
+#'
+#' @param ... Source(s) of known alive dates. One or more `lstalvdt_source()` objects are
+#'   expected.
+#'
+#' @param subject_keys Variables to uniquely identify a subject
+#'
+#'   A list of quosures where the expressions are symbols as returned by
+#'   `vars()` is expected.
+#'
+#' @author Stefan Bundfuss, Thomas Neitmann
+#'
+#' @return The input dataset with the `LSTALVDT` variable added.
+#'
+#' @keywords derivation adsl
+#'
+#' @export
+derive_var_lstalvdt <- function(dataset,
+                                ...,
+                                source_datasets,
+                                subject_keys = vars(STUDYID, USUBJID)) {
+  deprecate_warn("0.7.0", "derive_var_lstalvdt()", "derive_var_extreme_dt()")
+  derive_var_extreme_dt(
+    dataset,
+    new_var = LSTALVDT,
+    ...,
+    source_datasets = source_datasets,
+    mode = "last",
+    subject_keys = subject_keys)
+}
+
+#' Derive First or Last Datetime from Multiple Sources
```

Comment:
I think we would also want to use it to derive the first/last date from a single source?

Reply:
Yes, you are right we discussed this. However, I think we did not consider the `derive_vars_merged*()` functions. For a single source I would rather use these. I have added them to the "See also" section.

It is not ideal that the functionality of `derive_var_extreme_dt*()` and `derive_vars_merged*()` overlaps but I have no idea how we could change it.

Reply:
OK, let's keep this as is and see later if we can make this better.